examples/step-45/step-45.cc

```diff
@@ -14,417 +14,728 @@
+        for (unsigned int i=0; i<Utilities::MPI::n_mpi_processes(MPI_COMM_WORLD); ++i)
+          filenames.push_back (std::string("solution-") +
+                               Utilities::int_to_string (refinement_cycle, 2) +
+                               "." +
+                               Utilities::int_to_string(i, 2) +
+                               ".vtu");
+        const std::string
+        pvtu_master_filename = ("solution-" +
+                                Utilities::int_to_string (refinement_cycle, 2) +
+                                ".pvtu");
+        std::ofstream pvtu_master (pvtu_master_filename.c_str());
+        data_out.write_pvtu_record (pvtu_master, filenames);
+      }
+  }
 
-    std::ofstream output ("solution.vtk");
 
-    data_out.write_vtk (output);
-  }
 
+  template <int dim>
+  void
+  StokesProblem<dim>::refine_mesh ()
+  {
 
+    Vector<float> estimated_error_per_cell (triangulation.n_active_cells());
+
+    FEValuesExtractors::Scalar pressure(dim);
+    KellyErrorEstimator<dim>::estimate (dof_handler,
+                                        QGauss<dim-1>(degree+1),
+                                        typename FunctionMap<dim>::type(),
+                                        solution,
+                                        estimated_error_per_cell,
+                                        fe.component_mask(pressure));
+
+    parallel::distributed::GridRefinement::
+    refine_and_coarsen_fixed_number (triangulation,
+                                     estimated_error_per_cell,
+                                     0.3, 0.0);
+    triangulation.execute_coarsening_and_refinement ();
+  }
 
-  // @sect4{LaplaceProblem::run}
 
-  // And another function copied from previous programs:
-  void LaplaceProblem::run ()
+// @endcond
+//
+// @sect3{Setting up periodicity information on distributed triangulations}
+  template <int dim>
+  void StokesProblem<dim>::run ()
   {
-    make_grid_and_dofs();
-    assemble_system ();
-    solve ();
-    output_results ();
+    Point<dim> center;
+    const double inner_radius = .5;
+    const double outer_radius = 1.;
+
+    GridGenerator::quarter_hyper_shell (triangulation,
+                                        center,
+                                        inner_radius,
+                                        outer_radius,
+                                        0,
+                                        true);
+
+// Before we can use the constraints set above, we need to satisfy that all
+// the periodic faces are known to all the processes that own part of this
+// boundary are concerned with part of one of these. At this point we need
+// to think about how we want to prescribe periodicity. Therefore we have
+// to find the faces that should know about each other and tell the
+// triangulation about them. We do this in much the same manner as before.
+// The only change that we have to make is to change the type we save these
```

Comment:
This is a bit backward: where you refer to "before" you mean "above", but the code above is actually executed "after" the one here. It may be worth breaking out this part of the code into a `create_mesh()` function or similar.